app/scripts/metamask-controller.js

```diff
@@ -4845,13 +4868,6 @@ export default class MetamaskController extends EventEmitter {
    * Locks MetaMask
    */
   setLocked() {
-    const [trezorKeyring] = this.coreKeyringController.getKeyringsByType(
-      KeyringType.trezor,
-    );
-    if (trezorKeyring) {
-      trezorKeyring.dispose();
-    }
```

Comment:
In version `2.0.0` the `dispose` method has been renamed to `destroy`, so `KeyringController` is able to handle this internally.